inspectit-oce-agent/src/system-test/java/PrometheusMetricsExportSysTest.java

```diff
@@ -0,0 +1,55 @@
+import io.opencensus.stats.*;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PrometheusMetricsExportSysTest {
```

Comment:
Hmm, I am asking my self if this would be the way to go... OK so first we already tested the Prometheus exporter in the IntTest..

I would find it much better that we are not testing the specific exporter here, but using something generic like `opencensus-test` lib.. Because we can only test whatever exporter is there per default config and as they can be switched to one that are not testing.

I checked the lib and there is a support for the span exporting testing with https://github.com/census-instrumentation/opencensus-java/blob/master/testing/src/main/java/io/opencensus/testing/export/TestHandler.java.. There is no such thing for merics expoter, but I think this is because you don't need it.. You can simple get `Metrics.getExportComponent().getMetricProducerManager()` and iterate over all metrics with:

```
for (MetricProducer metricProducer : metricProducerManager.getAllMetricProducer()) {
        for (Metric metric : metricProducer.getMetrics()) {
```

What do you think, should we try this out.. The you can also easily assert of the properties (name, desc, unit) of your metric is correctly set and not only the responseText assertsion.